packages/web3-eth/types.ts

```diff
@@ -0,0 +1,160 @@
+import { HttpRpcResponse } from 'web3-providers-http/types';
+
+export type BlockIdentifier = number | 'latest' | 'earliest' | 'pending';
+
+export type EthLog = {
+    removed: boolean;
+    logIndex: string | null;
+    transactionIndex: string | null;
+    transactionHash: string | null;
+    blockHash: string | null;
+    blockNumber: string | null;
+    address: string;
+    data: string;
+    topics: string[];
+};
+
+export type EthTransaction = {
+    from: string;
+    to?: string;
+    gas?: BigInt;
+    gasPrice?: BigInt;
+    value?: BigInt;
+    data?: string;
+    nonce?: number;
+};
+
+export type EthMinedTransaction = {
+    blockHash: string | null;
+    blockNumber: string | null;
+    from: string;
+    gas: string;
+    gasPrice: string;
+    hash: string;
+    input: string;
+    nonce: string;
+    to: string | null;
+    transactionIndex: string | null;
+    value: string;
+    v: string;
+    r: string;
+    s: string;
+};
+
+export type EthBlock = {
+    number: string | null;
+    hash: string | null;
+    parentHash: string;
+    nonce: string | null;
+    sha3Uncles: string;
+    logsBloom: string | null;
+    transactionsRoot: string;
+    stateRoot: string;
+    receiptsRoot: string;
+    miner: string;
+    difficulty: string;
+    totalDifficulty: string;
+    extraData: string;
+    size: string;
+    gasLimit: string;
+    gasUsed: string;
+    timestamp: string;
+    transactions: EthMinedTransaction[];
+    uncles: string[];
+    root?: string;
+    status?: string;
+};
+
+export type EthTransactionReceipt = {
+    transactionHash: string;
+    transactionIndex: string;
+    blockHash: string;
+    blockNumber: string;
+    from: string;
+    to: string | null;
+    cumulativeGasUsed: string;
+    gasUsed: string;
+    contractAdress: string | null;
+    logs: EthLog[];
+    logsBloom: string;
+};
+
+export type EthCompiledSolidity = {
+    code: string;
+    info: {
+        source: string;
+        language: string;
+        languageVersion: string;
+        compilerVersion: string;
+        abiDefinition: any[];
+        userDoc: {
+            methods: { [key: string]: any };
+        };
+        developerDoc: {
+            methods: { [key: string]: any };
+        };
+    };
+};
+
+export type EthFilter = {
+    fromBlock?: BlockIdentifier;
+    toBlock?: BlockIdentifier;
+    address?: string;
+    topics?: string | null | string[][];
+    blochHash?: string;
+};
+
+export interface Web3EthOptions {
+    packageName?: string;
+    providerUrl: string;
+}
+
+export interface EthCallTransaction extends EthTransaction {
+    to: string;
+}
+
+export interface EthStringResult extends HttpRpcResponse {
```

Comment:
`extends HttpRpcResponse` will probably be changed to `RpcResponse` when writing `web3-providers-ws`, but leaving it for now

Reply:
Updated this in the `wyatt/rewrite/http-subscriptions` branch, but this needs to be merged first